src/components/token/Tokens.js

```diff
@@ -24,18 +27,101 @@ class Tokens extends React.Component {
             tokens: [],
             hasAfter: false,
             hasBefore: false,
-            searchText: ""
+            searchText: "",
+            sortBy: "uid",
+            order: "asc",
+            page: 1,
+            //This variable stores the pages and its searchAfter params that is passed to explorer-service
+            pageSearchAfter: [],
+            loading: true
         }
     }
 
-    onSearchButtonClicked = () => {
-        // TODO: Implement behavior when search button is clicked
+    componentDidMount = async() => {
+        //"Click" on search to make the first query
+        await this.onSearchButtonClicked();
+        this.setState({loading: false})
+    }
+
+    resetPageSearchAfter = () => {
+        this.setState({pageSearchAfter: [
+            {
+                page: 1,
+                searchAfter: []
+            }
+        ]})
+    }
+
+    getTokens = async (searchAfter) => {
+        const tokensRequest = await tokensApi.getList(this.state.searchText, this.state.sortBy, this.state.order, searchAfter)
```

Comment:
In a UX perspective, I believe we should start the loading icon every time we execute a request. The video seems good because the internet connection was fast but we should protect ourselves from a slow request and I think giving the user a UI feedback is good enough.

Reply:
Changed it. I've added loading icons. I will update on the next commit and also change the video on PR description